src/ancillary/__tests__/anchor-request-params-parser.test.ts

```diff
@@ -117,4 +136,14 @@ describe('AnchoRequestParamsParser', () => {
     expect(params.capCID).toEqual(FAKE_CAPCID)
   })
 
+  test('isleft indicates invalid car file', () => {
+    const validation =  parser.parse(CAR_FILE_INVALID as ExpReq)
+    expect(isLeft(validation)).toBeTruthy()
+  })
+
+  test('throws error if cannot decode car file', () => {
+    const validation =  parser.parse(CAR_FILE_NOT_BASE64URL as ExpReq)
+    console.log("a string of dashes, turns to " + JSON.stringify(validation))
```

Comment:
so validation of '---------' results in 

```
{"_tag":"Left","left":[{"trail":[{"key":"","type":{"name":"RequestAnchorParamsV2"},"actual":"----------------------------------------"}]}]}
```

I don't think then we should even check for https://github.com/ceramicnetwork/ceramic-anchor-service/blob/develop/src/ancillary/anchor-request-params-parser.ts#L99 if we never return any error no matter what?

Reply:
We **return** an error. We do not _throw_ it, as it breaks composability. If it returns Left value, then CAR file contains something unexpected.

There is a dozen reasons for an error there inside try-catch block apart just CAR file that is not spec-conformant. It could be an IPLD block of a weird structure, even if it is a pretty valid IPLD object. To maintain composability of codecs, we wrap a successful result or an error in an [Either](https://dev.to/attila_vecerek/the-either-monad-3ooc) monad.